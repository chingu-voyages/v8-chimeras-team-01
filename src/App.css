/* Ensure body and app root container truly cover 100% of the screen and that the body text is our selected font unless we change it specifically */

body {
  width: 100%;
  margin: 0;
  font-family: var(--font), sans-serif, arial;
}

/* Make sure React base element takes full width and height */

#root {
  position: absolute;
  width: 100%;
  height: 100%;
}


/* Remove decoration from links */

a:link,
a:visited,
a:hover,
a:active {
  text-decoration: none;
  color: inherit;
}


/* Add Theme Variables here (like colors and fonts we will use throughout the project) */

:root {
  --font: 'Roboto', sans-serif;
  --font-header: 'Roboto Condensed', sans-serif;
  --color-yellow: #FFED5C;
  --color-orange: #FFA324;
  --color-red: #FF4A24;
  --color-gray: #2D353D;
}

/* COMMON STYLES */

/* Flex-Box - Centered Column */
.fbc {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
}

/* pad:0 mar:0 */
.pm0 {
  padding: 0;
  margin: 0;
}


/* Setup Grid on top level div. This sets a space for the Nav and Footer and allows the center portion (1fr = all remaining space) to be used by whichever component is being shown. */

#top-level-container {
  display: grid;
  grid-template-columns: [c1] 100% [c2];
  grid-template-rows: [r1] 100px [r2] 1fr [r3] 200px [r4];
  width: 100%;
  height: auto;
  margin: 0;
  padding: 0;
}

/* FOOTER */
footer {
  grid-column: c1 / c2;
  grid-row: r3 / r4;
  width: 100%;
  margin-top: 1em;
  background-color: var(--color-gray);
  color: var(--color-orange);
  border-top: 3px solid #000;
}
