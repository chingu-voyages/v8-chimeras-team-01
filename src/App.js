import React, { Component } from 'react';
import { Switch, Route, Link } from 'react-router-dom';

//Stylesheets
import './App.css';

//Components
import Results from './components/Results';
import Questions from './components/Questions';
import LeaderBoard from './components/LeaderBoard';
import Landing from './components/Landing';
import Instructions from './components/Instructions';
import Games from './components/Games';
import Nav from './components/Nav';

class App extends Component {

  constructor(props){
    super(props);
    this.state = {
      gameList: [
        {
          id: "0001",
          name: "Waffles",
          image: <i class="fas fa-stroopwafel fa-6x"></i>,
          desc: "What you know 'bout Waffles??"
        },
        {
          id: "0002",
          name: "Cookies",
          image: <i class="fas fa-cookie fa-6x"></i>,
          desc: "You knwo you can eat them, but what else??"
        }
      ]
    }
  }

  render() {
    return (
      <div id="top-level-container">
        <Nav />

        <Switch>

          <Route exact path="/" component={Landing} />
          <Route path="/results" component={Results} />
          <Route path="/games"
                 render={(props) =>
                  <Games {...props}
                    gameList={this.state.gameList} />}
          />
          <Route path="/instructions" component={Instructions} />
          <Route path="/questions" component={Questions} />
          <Route path="/leaderBoard" component={LeaderBoard} />

        </Switch >

        <footer>
          <ul>
              <li><Link to="/">Landing Component</Link></li>
              <li><Link to="/results">Results Component</Link></li>
              <li><Link to="/games">Games</Link></li>
              <li><Link to="/questions">Questions Component</Link></li>
              <li><Link to="/leaderBoard">LeaderBoard Component</Link></li>
          </ul>
        </footer>
      </div>
    )
  }
}

export default App;
