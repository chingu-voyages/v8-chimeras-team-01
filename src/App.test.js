import React from 'react';
import { shallow } from 'enzyme';

import App from './App.js';


describe('App', () => {
  it('should render correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
  
});
