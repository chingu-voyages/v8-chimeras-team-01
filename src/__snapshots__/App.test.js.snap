// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  id="top-level-container"
>
  <_default />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/results"
    />
    <Route
      component={[Function]}
      path="/games"
    />
    <Route
      component={[Function]}
      path="/instructions"
    />
    <Route
      component={[Function]}
      path="/questions"
    />
    <Route
      component={[Function]}
      path="/leaderBoard"
    />
  </Switch>
  <footer>
    <ul>
      <li>
        <Link
          to="/"
        >
          Landing Component
        </Link>
      </li>
      <li>
        <Link
          to="/results"
        >
          Results Component
        </Link>
      </li>
      <li>
        <Link
          to="/games"
        >
          Games
        </Link>
      </li>
      <li>
        <Link
          to="/questions"
        >
          Questions Component
        </Link>
      </li>
      <li>
        <Link
          to="/leaderBoard"
        >
          LeaderBoard Component
        </Link>
      </li>
    </ul>
  </footer>
</div>
`;
